### PR TITLE
Add foreign keys to appeals and legacy_appeals tables (Phase 5c)

### PR DESCRIPTION
**Pull request by yoomlam**
Bumps #14732
* This is phase 5c of a [multi-phase plan](https://hackmd.io/wum2LtUfSHC-1o6-peBseQ) to add missing foreign keys.
* Phase 5b PR #16281.
* Phase 4 PR #16137.
* Phase 3 PR #16120.
* Phase 2 PR #16114.
* Phase 1 PR #16034 has more details.

### Description
Add foreign keys to appeals for the `dispatch_tasks`, `hearings`, and `legacy_hearings` tables. 

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
```ruby
Dispatch::Task.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=>0

LegacyHearing.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=>0

Hearing.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=> 0
```

Also check records in UAT as well.

### Database Changes

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

**Comment by yoomlam** [comment]
There's lots to fix In UAT:
```ruby
SpecialIssueList.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=> 4
Dispatch::Task.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=> 0
LegacyHearing.includes(:appeal).where.not(appeal_id: nil).select{|r| r.appeal == nil}.count
=> 606
```

File: spec/services/etl/legacy_hearing_syncer_spec.rb [hunk 1]
```diff
@@ -56,26 +56,6 @@
...
-
-        expect(LegacyHearing.count).to eq(3)
-        expect(ETL::LegacyHearing.count).to eq(2)
-        expect(etl_build_table).to be_complete
-        expect(etl_build_table.rows_rejected).to eq(1)
-        expect(etl_build_table.rows_inserted).to eq(2)
-        expect(etl_build_table.rows_updated).to eq(0)
-      end
-    end
-
```
Comment (yoomlam) [inline spec/services/etl/legacy_hearing_syncer_spec.rb:78]: With the foreign in place, there will no longer be orphaned LegacyHearing records.

**Review by wdrougas-tista — APPROVED** [review]
LGTM

**Comment by yoomlam** [comment]
[Slack](https://dsva.slack.com/archives/C013GETU45V/p1621864214023600)
Had to fix the migration in uat and preprod by
1. adding the missing (bad) migration file
2. rolling back
3. removing the (bad) migration file 
4. migrating (forward)

In preprod, there was an odd migration `201809112051441`, which I deleted.

**Pull request merged**